Docs/troubleshooting.md

```diff
@@ -0,0 +1,236 @@
+|Preferred receive window|Allows setting the device preferred receive window (RX1 or RX2). The default preferred receive window is 1| Add twin desired property `"PreferredWindow": 2` sets RX2 as preferred window. The absence of the twin property or setting the value to `1` will set RX1 as preferred window.|Using the second receive window increases the chances that the end application can process the upstream message and send a cloud to device message to the lora device without requiring and additional upstream message. Basically completing the round trip in less than 2 seconds.|
+
+**Important**: changes made to twin desired properties in devices that are already connected will only take effect once the network server is restarted or [cache is cleared](#cache-clearing).
+
+### Decoders
+
+The SensorDecoder tag is used to define which method will be used to decode the LoRa payload. If you leave it out or empty it will send the raw decrypted payload in the data field of the json message as Base64 encoded value to IoT Hub.
+
+If you want to decode it on the Edge you have the following two options:
+
+1. Specify a method that implements the right logic in the `LoraDecoders` class in the `LoraDecoders.cs` file of the `LoRaWan.NetworkServer`.
+
+2. Adapt the [DecoderSample](./Samples/DecoderSample) which allows you to create and run your own LoRa message decoder in an independent container running on your LoRa gateway without having to edit the main LoRa Engine. [This description](./Samples/DecoderSample#azure-iot-edge-lorawan-starter-kit) shows you how to get started.
+
+In both cases, we have already provided a simple decoder called `"DecoderValueSensor"` that takes the whole payload as a single numeric value and constructs the following json output as a response (The example of an Arduino sending a sensor value as string (i.e. "23.5") is available in the [Arduino folder](./Arduino)):
+
+```json
+{
+  .....
+    "data": {"value": 23.5}
+  .....
+}
+```
+
+To add the sample `"DecoderValueSensor"` to the sample LoRa device configured above, change it's desired properties in IoT Hub as follows for option 1:
+
+```json
+"desired": {
+    "AppEUI": "App EUI",
+    "AppKey": "App Key",
+    "GatewayID": "",
+    "SensorDecoder": "DecoderValueSensor"
+  },
+```
+
+or as follows for option 2:
+
+```json
+"desired": {
+    "AppEUI": "App EUI",
+    "AppKey": "App Key",
+    "GatewayID": "",
+    "SensorDecoder": "http://your_container_name/api/DecoderValueSensor"
+  },
+```
+
+The `"DecoderValueSensor"` decoder is not a best practice but it makes it easier to experiment sending sensor readings to IoT Hub without having to change any code.
+
+if the SensorDecoder tag has a "http" in it's string value, it will forward the decoding call to an external decoder, as described in option 2 above, using standard Http. The call expects a return value with the same format as the json here above or an error string.
```

Comment:
Capitalize the word "if". 

I dont understand the last sentence as to which json is being referred to... it says "here above"... If you mean the json code above, then delete the work "here" . --> The call expects a return value with the same format as the json above or an error string.